core/tests/unit/ldml/ldml_test_source.cpp

```diff
@@ -126,7 +126,9 @@ bool LdmlTestSource::get_expected_beep() const {
 // trim from start (in place)
 static inline void
 ltrim(std::string &s) {
-  s.erase(s.begin(), std::find_if(s.begin(), s.end(), [](int ch) { return !std::isspace(ch); }));
+  // std::isspace chokes on negative (!) ints here, so spare it the trouble if ch is negative
+  // (possibly unsigned char to integer cast?)
+  s.erase(s.begin(), std::find_if(s.begin(), s.end(), [](int ch) { return (ch < 0) || !std::isspace(ch); }));
```

Comment:
k_213 contains some random chars. It was a 'scratch' area I was using while editing. This fixes a crash in std::isspace.

Reply:
I'm a little concerned that we are missing something more important. How can ch be `< 0`?

Reply:
Codepoint U+00AB in UTF-8 is `C2 AB`, and 0xC2 as a signed char cast to integer is `-62`

Reply:
![image](https://github.com/keymanapp/keyman/assets/855219/bf621a1e-ded0-4d2b-8dad-b39a8379cfa8)

Reply:
clarified, PTAL.

(Won't be hit if we drop K213)

Reply:
std::ifstream and std::string by default seem to read in utf-8 as an array of chars.

Reply:
see parent comment- i have a commit ready that drops K213 and also removes the lstrip() code, then we can revist that in 18